kubernetes/data_source_kubernetes_ingress_test.go

```diff
@@ -62,6 +62,48 @@ func TestAccKubernetesDataSourceIngress_basic(t *testing.T) {
 	})
 }
 
+func TestAccKubernetesDataSourceIngress_stateUpgradeV0_loadBalancerIngress(t *testing.T) {
+	name := fmt.Sprintf("tf-acc-test-%s", acctest.RandStringFromCharSet(10, acctest.CharSetAlphaNum))
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t); skipIfNotRunningInEks(t) },
+		ExternalProviders: testAccExternalProviders,
+		CheckDestroy:      testAccCheckKubernetesIngressDestroy,
+		Steps: []resource.TestStep{
+			{ // Create resource using schema v0.
+				Config: requiredProviders() + testAccKubernetesDataSourceIngressConfig_stateUpgradev0("kubernetes-released", name),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr("kubernetes_ingress.test", "metadata.0.name", name),
+				),
+			},
+			{ // Create data source using schema v0.
+				Config: requiredProviders() + testAccKubernetesDataSourceIngressConfig_stateUpgradev0("kubernetes-released", name) +
+					testAccKubernetesDataSourceIngressConfig_read(),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr("data.kubernetes_ingress.test", "metadata.0.name", name),
+					resource.TestCheckResourceAttrSet("data.kubernetes_ingress.test", "load_balancer_ingress.0.hostname"),
+				),
+			},
+			{ // Apply StateUpgrade to resource. This will delete the data source, to be re-read on next apply.
+				Config: requiredProviders() + testAccKubernetesDataSourceIngressConfig_stateUpgradev0("kubernetes-local", name) +
+					testAccKubernetesDataSourceIngressConfig_read(),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttrSet("kubernetes_ingress.test", "status.0.load_balancer.0.ingress.0.hostname"),
+					resource.TestCheckNoResourceAttr("kubernetes_ingress.test", "load_balancer_ingress.0.hostname"),
+				),
+			},
+			{ // Re-populate the data source with the StateUpgraded resource data.
+				Config: requiredProviders() + testAccKubernetesDataSourceIngressConfig_stateUpgradev0("kubernetes-local", name) +
+					testAccKubernetesDataSourceIngressConfig_read(),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckNoResourceAttr("data.kubernetes_ingress.test", "load_balancer_ingress.0.hostname"),
+					resource.TestCheckResourceAttrSet("data.kubernetes_ingress.test", "status.0.load_balancer.0.ingress.0.hostname"),
+				),
+			},
+		},
+	})
+}
+
```

Comment:
I think there must be some refresh mechanism missing from the acceptance testing framework, because the data source never seems to refresh. Even after multiple applies, it shows the data present in the old location (`load_balancer_ingress` instead of `status.0.load_balancer.0.ingress`).

However, in manual testing, this works as expected. The data source is able to read data from the migrated resource, and it looks correct in state.

I might just want to delete this test, since it doesn't reflect the actual state of things outside of the test environment.

```
$ TESTARGS="-run 'TestAccKubernetesDataSourceIngress_stateUpgradeV0_loadBalancerIngress'" make testacc
...

=== RUN   TestAccKubernetesDataSourceIngress_stateUpgradeV0_loadBalancerIngress
    data_source_kubernetes_ingress_test.go:68: Step 4/4 error: Check failed: 2 errors occurred:
                * Check 1/2 error: data.kubernetes_ingress.test: Attribute 'load_balancer_ingress.0.hostname' found when not expected
                * Check 2/2 error: data.kubernetes_ingress.test: Attribute 'status.0.load_balancer.0.ingress.0.hostname' expected to be set
```